src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.DukeException;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.common.InputChecker;
+import duke.ui.BotUI;
+
+/**
+ * Deals with making sense of the user input.
+ * Contains methods that return Command for execution in Duke class and extracting information from user's raw input.
+ */
+
+public class Parser {
+
+    private static final BotUI UI = new BotUI();
+
+    private static String[] splitInput(String input, String regex) {
+        return input.split(regex, 2);
+    }
+
+    /**
+     * Returns different type of Command according to the user raw input.
+     *
+     * @param rawInput user's raw input.
+     * @return Command to be executed in Duke class.
+     * @throws DukeException - thrown if user command is invalid.
+     */
+    public static Command parse(String rawInput) throws DukeException {
+        try {
+            String[] commandAndDetail = rawInput.split(" ", 2);
+            String command = commandAndDetail[0];
+            if (commandAndDetail.length < 2) {
+                if (command.equals("list")) {
+                    return new ListCommand(command);
+                } else if (command.equals("bye")) {
+                    return new ExitCommand(command);
+                }
+            }
+            String detail = commandAndDetail[1];
+            if (command.equals("todo") || command.equals("event") || command.equals("deadline")) {
+                InputChecker.checkInput(rawInput);
+                return new AddCommand(command, detail);
+            } else if (command.equals("mark") || command.equals("unmark")) {
+                return new MarkCommand(command, detail);
+            } else if (command.equals("delete")) {
+                return new DeleteCommand(command, detail);
+            } else if (command.equals("find")) {
+                return new FindCommand(command, detail);
+            } else {
+                throw new DukeException(UI.invalidFormat());
+            }

```

Comment:
Code is well formatted and the abstraction makes it easy to read. Would recommend you to explore using `switch case` syntax as well